pylint/checkers/stdlib.py

```diff
@@ -599,6 +620,40 @@ def _check_lru_cache_decorators(self, decorators: nodes.Decorators) -> None:
                 confidence=interfaces.INFERENCE,
             )
 
+    def _check_dispatch_decorators(self, node: nodes.FunctionDef) -> None:
+        decorators_names_set: set[str] = set()
+        decorators_name_node_map: dict[str, nodes.NodeNG] = {}
```

Comment:
Can we making the mapping a tuple so we only have two temp vars?

I was also thinking of using `if "singledispatch" in decorators_map or "classmethod" in decorators` but I'm not sure that's more readable.